src/EnergyPlus/HVACHXAssistedCoolingCoil.cc

```diff
@@ -611,12 +611,14 @@ namespace HVACHXAssistedCoolingCoil {
         // Get the data for the Coil:Water:CoolingHeatExchangerAssisted objects
         CurrentModuleObject = "CoilSystem:Cooling:Water:HeatExchangerAssisted";
 
-        for (HXAssistedCoilNum = NumHXAssistedDXCoils + 1; HXAssistedCoilNum <= NumHXAssistedWaterCoils; ++HXAssistedCoilNum) {
+        for (HXAssistedCoilNum = NumHXAssistedDXCoils + 1; HXAssistedCoilNum <= state.dataHVACAssistedCC->TotalNumHXAssistedCoils;
+             ++HXAssistedCoilNum) {
+            int thisWaterHXNum = HXAssistedCoilNum - NumHXAssistedDXCoils;
```

Comment:
This is the 2nd loop in getInput. If the input file had both flavors on HXAssisted coil, this loop was skipped. For example if there were one of each flavor the DX version would get read in and then here the Water version would be skipped.

    for (HXAssistedCoilNum = 2, HXAssistedCoilNum < 1; ++ HXAssistedCoilNum)